azurerm/internal/services/storage/storage_account_resource.go

```diff
@@ -1687,6 +1888,90 @@ func expandStaticWebsiteProperties(input []interface{}) accounts.StorageServiceP
 	return properties
 }
 
+func flattenArmStorageAccountAzureFilesIdentityBasedAuthentication(input *storage.AzureFilesIdentityBasedAuthentication) []interface{} {
+	if input == nil || input.DirectoryServiceOptions == storage.DirectoryServiceOptionsNone {
+		return make([]interface{}, 0)
+	}
+
+	var directoryServiceOptions storage.DirectoryServiceOptions
+	if input.DirectoryServiceOptions != "" {
+		directoryServiceOptions = input.DirectoryServiceOptions
+	}
+	return []interface{}{
+		map[string]interface{}{
+			"directory_service_options": directoryServiceOptions,
```

Comment:
You can shorten this
```suggestion
	return []interface{}{
		map[string]interface{}{
			"directory_service_options": input.DirectoryServiceOptions,
```